service/history/workflowTaskHandler.go

```diff
@@ -436,41 +427,37 @@ func (handler *workflowTaskHandlerImpl) handleCommandFailWorkflow(
 		return nil
 	}
 
-	// below will check whether to do continue as new based on backoff & backoff or cron
+	// First check retry policy to do a retry.
 	backoffInterval, retryState := handler.mutableState.GetRetryBackoffDuration(attr.GetFailure())
-	continueAsNewInitiator := enumspb.CONTINUE_AS_NEW_INITIATOR_RETRY
-	// first check the backoff retry
-	if backoffInterval == backoff.NoBackoff {
-		// if no backoff retry, set the backoffInterval using cron schedule
-		backoffInterval, err = handler.mutableState.GetCronBackoffDuration()
+	if backoffInterval != backoff.NoBackoff {
+		_, err := handler.mutableState.AddFailWorkflowEvent(
```

Comment:
same here, maybe consolidate the cron / retry logic in one place and here just blindly do `fail workflow`?
e.g. 
* get backoff / retry 
* fail workflow
* check backoff / retry and do cron / retry accordingly